app/models/user.rb

```diff
@@ -75,6 +76,14 @@ def remove_devise_validations
     end
   end
 
+  def active_session
+    sessions.find_by(status: 'in-progress')
```

Comment:
lets memoize that since we are calling it many times(at least 2) in the controller.

```suggestion
    @active_session ||= sessions.find_by(status: 'in-progress')
```